File: front-end/src/dto/serverPayloads/CreatureStunnedPayload.js
import { Creature } from "../../models/Creature";

export class CreatureStunnedPayload {

    constructor(creature, damage) {
        this.creature = creature
    }

    static fromDict(data) {
        return new CreatureStunnedPayload(
            Creature.fromDict(data.creature)
        );
    }   
}

File: server/s2game/src/main/java/com/spectopics/s2game/dto/serverPayloads/CreatureStunnedPayload.java
package com.spectopics.s2game.dto.serverPayloads;

import com.spectopics.s2game.models.Creature;

public class CreatureStunnedPayload {
    // Tells the clients that a creature has been stunned.
    // They loose their turn.
    public Creature creature;

    public CreatureStunnedPayload(Creature creature) {
        this.creature = creature;
    }
}
